### PR TITLE
Make exosuit give points

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
Make the exosuits give points when building them.
Ripley give 10k.
Firefighter give 15k.
Oddysseus give 17.5k.
Durand give 20k.
Gygax give 25k.
Phazon give 50k.

### CODE REVIEW

File: code/datums/helper_datums/construction_datum.dm [hunk 1]
```diff
@@ -2,115 +2,138 @@
...
+/datum/construction/proc/check_all_steps(atom/used_atom, mob/user as mob) //check all steps, remove matching one.
+	for(var/i=1;i<=steps.len;i++)
+		var/list/L = steps[i];
+		if(istype(used_atom, L["key"]))
+			if(custom_action(i, used_atom, user))
+				steps[i]=null;//stupid byond list from list removal...
+				listclearnulls(steps);
+				if(!steps.len)
+					spawn_result()
 				return 1
```
Comment (Trilbyspaceclone) [inline code/datums/helper_datums/construction_datum.dm:68]:
```suggestion
			return 1
```

File: code/datums/helper_datums/construction_datum.dm [hunk 2]
```diff
@@ -2,115 +2,138 @@
...
+	if(ispath(L["backkey"]))
+		if(L["backkey"] && istype(used_atom, L["backkey"]))
+			return BACKWARD //to the last step -> backwards
+	else
+		possibleWays[L["backkey"]] = BACKWARD
+	if(istype(used_atom, /obj/item))
+		var/obj/item/I = used_atom
+		var/selected = I.get_tool_type(user, possibleWays, holder)
+		return selected && possibleWays[selected]
 		return FALSE
```
Comment (Trilbyspaceclone) [inline code/datums/helper_datums/construction_datum.dm:128]:
```suggestion
	return FALSE
```

File: code/datums/helper_datums/construction_datum.dm [hunk 3]
```diff
@@ -2,115 +2,138 @@
...
+			for(var/obj/machinery/computer/rdconsole/RD in GLOB.computer_list) // Check every RnD computer in existance
+				if(RD.id == 1) // only core gets the science
+					RD.files.research_points += give_points // Give the points
+					var/obj/item/device/radio/R = new /obj/item/device/radio // New radio to send a message
+					R.channels = list("Science" = 1)
+					R.autosay("Exosuit constructed, granting [give_points] research points", "Legio Cybernetica's Announcement System" ,"Science") // Message on the Science channel using the radio
+					qdel(R)
+
+	else
 		set_desc(steps.len)
```
Comment (Trilbyspaceclone) [inline code/datums/helper_datums/construction_datum.dm:36]:
```suggestion
	set_desc(steps.len)
```

**Comment by Trilbyspaceclone** [comment]
i would make riply 10k
figherfighter 15k
oddy 17.5k
durhand 20k
glaxy 25k
and phazon 50k

**Comment by r4d6** [comment]
> i would make riply 10k
> figherfighter 15k
> oddy 17.5k
> durhand 20k
> glaxy 25k
> and phazon 50k

Done

**Comment by ThePainkiller** [comment]
So you made it work this time? This is a great boon!

**Pull request merged**